src/custom_token.rs

```diff
@@ -0,0 +1,554 @@
+// Not public API.
+#[doc(hidden)]
+#[macro_export(local_inner_macros)]
+macro_rules! punct_len {
+    (+)      => ( 1usize );
+    (+=)     => ( 2usize );
+    (&)      => ( 1usize );
+    (&&)     => ( 2usize );
+    (&=)     => ( 2usize );
+    (@)      => ( 1usize );
+    (!)      => ( 1usize );
+    (^)      => ( 1usize );
+    (^=)     => ( 2usize );
+    (:)      => ( 1usize );
+    (::)     => ( 2usize );
+    (,)      => ( 1usize );
+    (/)      => ( 1usize );
+    (/=)     => ( 2usize );
+    (.)      => ( 1usize );
+    (..)     => ( 2usize );
+    (...)    => ( 3usize );
+    (..=)    => ( 3usize );
+    (=)      => ( 1usize );
+    (==)     => ( 2usize );
+    (>=)     => ( 2usize );
+    (>)      => ( 1usize );
+    (<=)     => ( 2usize );
+    (<)      => ( 1usize );
+    (*=)     => ( 2usize );
+    (!=)     => ( 2usize );
+    (|)      => ( 1usize );
+    (|=)     => ( 2usize );
+    (||)     => ( 2usize );
+    (#)      => ( 1usize );
+    (?)      => ( 1usize );
+    (->)     => ( 2usize );
+    (<-)     => ( 2usize );
+    (%)      => ( 1usize );
+    (%=)     => ( 2usize );
+    (=>)     => ( 2usize );
+    (;)      => ( 1usize );
+    (<<)     => ( 2usize );
+    (<<=)    => ( 3usize );
+    (>>)     => ( 2usize );
+    (>>=)    => ( 3usize );
+    (*)      => ( 1usize );
+    (-)      => ( 1usize );
+    (-=)     => ( 2usize );
+    (~)      => ( 1usize );
+    ($tt:tt) => ( my_compile_error!("Punctuation `{}` is not supported", stringify!($tt)) );
```

Comment:
Now it says:

```console
error: compile_error! takes 1 argument
 --> src/main.rs:2:5
  |
2 |     syn::custom_punctuation!(LeftRightArrow, <x>);
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Reply:
I gave up trying to be too helpful and just gave a simple enough error message